instaclustr/resource_firewall_rule.go

```diff
@@ -23,7 +23,11 @@ func resourceFirewallRule() *schema.Resource {
 
 			"rule_cidr": {
```

Comment:
[desired] this could def cause some confusion if both marked as optional ? ie it could pass the terraform validation yet fail at the API end.  I see we have tried todo that logic within the `resourceFirewallRuleCreate` call.  Would it be cleaner todo it within a custom `ValidateFunc` ? and thus catch it before proceeding into the api calling code -> https://github.com/hashicorp/terraform-plugin-sdk/blob/9f0df37a8fdb2627ae32db6ceaf7f036d89b6768/helper/schema/schema.go#L310-L312

Reply:
@arjunrajshekhar  did raise that, but then pointed out you can only validate an individual field, not validate across multiple fields? 
Happy to refactor if thats incorrect.